### ISSUE TITLE
RF02 triggered when passing a table as an argument to a function

### ISSUE THREAD

**Issue by CRBl69** [sqlfluff/sqlfluff#5953]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When passing a table as an argument to a function, RF02 is reported.

### Expected Behaviour

RF02 should not be reported, as the table does not need a qualifier.

### Observed Behaviour

```
L:  16 | P:  28 | RF02 | Unqualified reference 'mytable' found in select with
                       | more than one referenced table/view.
                       | [references.qualification]
```

### How to reproduce

```sql
CREATE TABLE mytable (
    a int,
    b int
);

CREATE TABLE othertable (
    c int
);

CREATE FUNCTION myadd(mytable)
RETURNS NUMERIC AS $$
  SELECT $1.a + $1.b;
$$ IMMUTABLE LANGUAGE sql;

CREATE VIEW myview AS
SELECT othertable.c, myadd(mytable) FROM mytable, othertable;
```

### Dialect

PostgreSQL

### Version

sqlfluff, version 3.0.4

### Configuration

[sqlfluff]

dialect = postgres
templater = raw

max_line_length = 120

processes = -1

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by swastim01** [comment]
I tested the reproduction on the current `main` branch (commit `3f105a2b23a7f175a41b2d154605231a6b0ce294`) using:

```bash
sqlfluff lint test.sql --dialect postgres
```

with the SQL from this issue, and I was **not** able to reproduce the reported `RF02` violation. The only violations reported were unrelated (`CP05`, `LT09`, `AL03`, and `LT12`).

Unless I'm missing an additional configuration or a different reproduction case, it appears this issue no longer reproduces on the current `main` branch.

Could a maintainer confirm whether this was fixed as part of another change? If so, it may be appropriate to close this issue.

**Comment by be-student** [comment]
I’m working on a focused fix and regression test for this RF02 false positive.

**Comment by be-student** [comment]
Correction: the reported query already passes RF02 on current `main` (`eb38850aa`), so no patch is needed. I’m releasing this.
